app-search.js

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env node
```

Comment:
Something is missing here. When would app-search.js be used? Is this intended to replace app.js? I don't see a point in a breaking change to the file structure without some justification.